squirrel.frontier/src/main/java/org/dice_research/squirrel/frontier/impl/FrontierQueryGenerator.java

```diff
@@ -0,0 +1,212 @@
+            stringBuilder.append("> { ");
+        }
+        for (Triple triple : listBufferedTriples) {
+            stringBuilder.append(formatNodeToString(triple.getSubject()));
+            stringBuilder.append(formatNodeToString(triple.getPredicate()));
+            stringBuilder.append(formatNodeToString(triple.getObject()));
+            stringBuilder.append(". ");
+        }
+        if (!defaultGraph) {
+            stringBuilder.append("} ");
+        }
+        stringBuilder.append("}");
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Return a time stamp query for the default graph.
+     * It will return triples with time stamp contained in the default graph.
+     * @return All triples with time stamp in the default graph.
+     */
+  
+    public Query getOutdatedUrisQuery() {
+        return getOutdatedUrisQuery(null, true);
+    }
+    public Query getOutdatedUrisQuery(String graphID, boolean defaultGraph) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("PREFIX  sq:   <http://w3id.org/squirrel/vocab#>\n" + 
+        		"PREFIX  prov: <http://www.w3.org/ns/prov#>\n" + 
+        		"PREFIX  xsd:  <http://www.w3.org/2001/XMLSchema#>"
+        		+ "SELECT ?uri  WHERE { \n ");
+        // + "SELECT ?uri  WHERE { \n ");
+        if (!defaultGraph) {
+            stringBuilder.append("GRAPH <");
+            stringBuilder.append(graphID);
+            stringBuilder.append("> { ");
+        }
+        stringBuilder.append("{\n" + 
+        		"SELECT ?uri ?endtime (NOW() - (?endtime) AS ?diff)\n" + 
+        		"WHERE{\n" + 
+        		"\n" + 
+        		"  {\n" + 
+        		"    SELECT  ?uri  (MAX(?timestamp) as ?endtime)\n" + 
+        		"    WHERE\n" + 
+        		"    { \n" + 
+        		"        ?s  sq:crawled  ?uri ;\n" + 
+        		"        prov:endedAtTime  ?timestamp.\n" + 
+        		"\n" + 
+        		"    }\n" + 
+        		"    GROUP BY ?uri\n" + 
+        		"  } \n" + 
+        		"}\n" + 
+        		"}\n" + 
+        		"FILTER(?diff > \"18000\"^^xsd:double)\n" +
+        		"");
+        if (!defaultGraph) {
+            stringBuilder.append("}");
+        }
+       
+       // stringBuilder.append("}GROUP BY ?uri");
+          stringBuilder.append("}");
+       
+        Query query = QueryFactory.create(stringBuilder.toString());
+        return query;
+    }
+
+    public Query getSelectQuery() {
+        return getSelectQuery(null, true);
+    }
+    /**
+     * Return a select query for the given graphID or default graph.
+     * It will return all triples contained in the graph.
+     * @return All triples contained in the default graph.
+     * @param graphID      The id of the graph from which you want to select.
+     * @param defaultGraph Identify if query is for the default graph
+     * @return All triples contained in the graph.
+     */
+    public Query getSelectQuery(String graphID, boolean defaultGraph) {
```

Comment:
The same as above. Your call in line 134 already shows that you would use `null` instead of a URI when you would like to use the default graph.

Apart from that, the method name is very generic. You may want to change it to a name that expresses what the query actually selects.

Is the method even used somewhere? If not, it should be removed.